dipy/tracking/utils.py

```diff
@@ -412,6 +412,145 @@ def seeds_from_mask(mask, density=[1, 1, 1], voxel_size=None, affine=None):
     return seeds
 
 
+def seeds_from_mesh(vertices, faces, n_samples=None, affine=None):
+    """
+    Creates randomly sampled seeds for fiber tracking from a surface mesh.
+    Seed points are uniformly sampled from triangles in the mesh.
+
+    Parameters
+    ----------
+    vertices : 2d float array
+        3d coordinates of vertices in a surface mesh
+    faces : 2d int array
+        array of triangles in a surface mesh, where each triangle is a list
+        of vertex indices in the triangle
+    n_samples : int
+        number of samples per face
+    affine : array, (4, 4)
+        The mapping between voxel indices and the point space for seeds. A
+        seed point at the center the voxel ``[i, j, k]`` will be represented as
+        ``[x, y, z]`` where ``[x, y, z, 1] == np.dot(affine, [i, j, k , 1])``.
+    
+    Returns
+    -------
+    seeds : 2d float array
+        3d coordinates of seeds sampled from each triangle in a mesh
+    assignment : 2d int array
+        closest vertex in original mesh to each seed
+
+    Examples
+    --------
+    Note:  There is some stochasticity involved in the seed generation process,
+    as points are sampled randomly from a uniform distribution.
+
+    >>> vertices = np.asarray([[1, 0, 0],
+                               [0, 1, 0],
+                               [0, 0, 1]])
+    >>> faces = np.asarray([0,1,2])[:,None]
+    >>> n_samples = 3
+    >>> [seeds, assignment] = seeds_from_mesh(vertices, faces, n_samples, affine=None)
+    >>> print(seeds) 
+    >>> array([[0.14434192, 0.3372361 , 0.51842198],
+       [0.31493263, 0.68194987, 0.0031175 ],
+       [0.47816217, 0.49051105, 0.03132679]])
+    >>> print(assignment)
+    >>> [2, 1, 1]
+
+    """
+
+    # If n_samples is unspecified, samples are the original vertex coordinates
+    if not n_samples:
+        seeds = vertices
+        assignment = np.arange((len(vertices)))
+    
+    else:
+
+        if isinstance(faces, np.ndarray):
+            faces = faces.T.tolist()
+
+        S = len(faces)*n_samples
+
+        seeds = np.zeros((S, 3))
+        assignment = np.zeros((S, ))
+
+        for c, face in enumerate(faces):
+
+            # Get coordinates of each vertex in triangle
+            coords = vertices[face, :]
+            # Generate seeds from triangle and compute nearest vertex
+            face_seeds, nearest = sample_triangle(coords, face, n_samples)
+
+            lo = c*n_samples
+            hi = (c+1)*n_samples
```

Comment:
[PEP8](https://www.python.org/dev/peps/pep-0008/#whitespace-in-expressions-and-statements). I'm not sure about DiPy rules, but there's almost always spaces around operators.